loopy/target/c/__init__.py

```diff
@@ -692,7 +692,14 @@ def with_types(self, arg_id_to_dtype, callables_table):
                 raise LoopyTypeError(f"'{name}' needs order to be an int-type.")
 
             if arg_id_to_dtype[1].numpy_dtype == np.float32:
-                name_in_target = name[-2:]+"f"
+                # See e.g.
+                # https://github.com/flang-compiler/flang/blob/master/runtime/libpgmath/lib/common/mthdecls.h
+                # for Bessel function names
+                import os
+                if os.uname().sysname == "Linux":
```

Comment:
Is this a good heuristic to use? AFAIK, the GNU compiler is perfectly usable on the Mac, it's just that Apple chooses to no longer include them by default.

Reply:
I _think_ this is independent from the compiler and depends only on the OS. E.g., on MacOS, `math.h` only has `yn` etc., not `ynf`. On my system, this is `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/math.h`

Reply:
Looks like you're right! Thanks for explaining.